src/plugins/workspace/server/plugin.ts

```diff
@@ -115,12 +116,44 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
       if (path === '/') {
         const workspaceListResponse = await this.client?.list(
           { request, logger: this.logger },
-          { page: 1, perPage: 1 }
+          { page: 1, perPage: 100 }
         );
+        const basePath = core.http.basePath.serverBasePath;
+
         if (workspaceListResponse?.success && workspaceListResponse.result.total > 0) {
-          return toolkit.next();
+          const workspaceList = workspaceListResponse.result.workspaces;
+          // If user only has one workspace, go to overview page of that workspace
+          if (workspaceList.length === 1) {
+            return response.redirected({
+              headers: {
+                location: `${basePath}/w/${workspaceList[0].id}/app/${WORKSPACE_NAVIGATION_APP_ID}`,
+              },
+            });
+          }
+          const [coreStart] = await core.getStartServices();
+          const uiSettings = coreStart.uiSettings.asScopedToClient(
+            coreStart.savedObjects.getScopedClient(request)
+          );
+          // Temporarily use defaultWorkspace as a placeholder
+          const defaultWorkspaceId = await uiSettings.get('defaultWorkspace');
+          const defaultWorkspace = workspaceList.find(
+            (workspace) => workspace.id === defaultWorkspaceId
+          );
+          // If user has a default workspace configured, go to overview page of that workspace
+          // If user has more than one workspaces, go to homepage
+          if (defaultWorkspace) {
+            return response.redirected({
+              headers: {
+                location: `${basePath}/w/${defaultWorkspace.id}/app/${WORKSPACE_NAVIGATION_APP_ID}`,
+              },
+            });
+          } else {
+            return response.redirected({
+              headers: { location: `${basePath}/app/home` },
```

Comment:
i see this logic depends on `core.http.registerOnPostAuth`, how about if security is disabled case, will it work?

Reply:
Although the security plugin is disable, the `core.http.registerOnPostAuth` will also be triggered.